book.rb

```diff
@@ -6,12 +7,29 @@ def initialize(title, author)
   end
   attr_accessor :title, :author, :rentals
 
-  def self.all
-    ObjectSpace.each_object(self).to_a
-  end
-
   def add_rental(rental)
     @rentals.push(rental)
     book.rental = self
   end
+
+  def tohash
+    temp_hash = {
+      @title => {
+        'title' => @title,
+        'author' => @author,
+        'rentals' => @rentals
+      }
+    }
+    tojson(temp_hash)
+  end
+
+  def tojson(temp_hash)
+    if File.zero?('books.json')
+      File.write('books.json', JSON.pretty_generate([temp_hash]))
+    else
+      data_from_json = JSON.parse(File.read('books.json'))
+      data_from_json = [data_from_json] if data_from_json.class != Array
+      File.write('books.json', JSON.pretty_generate(data_from_json << temp_hash))
+    end
```

Comment:
- [ ] The book data are saved directly after creating a new book. As per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_preserve_data.md#:~:text=On%20exit%20from%20your%20app%20save%20all%20the%20data%20(books%2C%20people%2C%20and%20rentals)%20into%20JSON%20files.), Please save this data on exit only.